lib/settings.rb

```diff
@@ -29,27 +29,20 @@ def initialize(host, username, password)
   module_function
   filepath         = File.expand_path('../config/settings.yml', __dir__)
   @h               = YAML::load_file(filepath)
+  @config_keys     = @h.keys.map { |k| k.to_sym }
   @pq_rest_api     = PqRestApi.from_env
 
   def pq_rest_api
     @pq_rest_api
   end
 
-  def mail_from
-    @h['mail_from']
-  end
-
-  def mail_reply_to
-    @h['mail_reply_to']
-  end
-
-  def mail_tech_support
-    @h['mail_tech_support']
-  end
-
-  def http_client_timeout
-    @h['http_client_timeout']
-  end
+  def method_missing(method, *params)
+    if @config_keys.include?(method)
```

Comment:
Nice one for trying to abstract things a bit more, but I am not sure this is the right place and the right way. There are two reasons why I am personally not a big fan of the  `method_missing?` approach:

- It doesn't play nicely with tools: breaks autocomplete in both REPL and editors.
- it doesn't play nicely with documentation tools like RDoc or YARD. 
- It makes things more dry, but also less explicit.

In this specific case, I do not really want to generate a method for each key found in the config file. Quite the opposite, the whole purpose of this Settings module is to ensure that the YAML file has the expected fields.